src/main/java/bio/terra/workspace/db/DataReferenceDao.java

```diff
@@ -56,7 +63,13 @@ public String createDataReference(
     paramMap.put("credential_id", credentialId);
     paramMap.put("resource_id", resourceId == null ? null : resourceId.toString());
     paramMap.put("reference_type", referenceType == null ? null : referenceType.toString());
-    paramMap.put("reference", reference);
+    try {
+      paramMap.put("reference", objectMapper.writeValueAsString(reference));
+    } catch (JsonProcessingException e) {
+      // TODO: add logger and print out the reference
+      throw new InvalidDataReferenceException(
+          "Couldn't convert reference to JSON. This... shouldn't happen.");
```

Comment:
this could easily happen, since it is user-supplied input, right?

Reply:
It shouldn't, we should already have validated it and checked that the reference exists by calling `validateReference`.